eng/Directory.Build.Data.props

```diff
@@ -45,6 +45,11 @@
     <IsTestSupportProject Condition="'$(IsTestProject)' != 'true' and ($(MSBuildProjectDirectory.Contains('/tests/')) or $(MSBuildProjectDirectory.Contains('\tests\')))">true</IsTestSupportProject>
     <EnableClientSdkAnalyzers Condition="'$(IsClientLibrary)' == 'true' and '$(IsTestProject)' != 'true' and '$(IsTestSupportProject)' != 'true' and '$(IsSamplesProject)' != 'true'">true</EnableClientSdkAnalyzers>
     <EnableFxCopAnalyzers Condition="'$(IsClientLibrary)' == 'true' and '$(IsTestProject)' != 'true' and '$(IsTestSupportProject)' != 'true' and '$(IsSamplesProject)' != 'true'">true</EnableFxCopAnalyzers>
+    <EnableStyleCopAnalyzers Condition="'$(EnableStyleCopAnalyzers)' == '' and '$(IsClientLibrary)' == 'true'">true</EnableStyleCopAnalyzers>
+  </PropertyGroup>
+
+  <!-- CodeAnalysis RuleSet -->
+  <PropertyGroup Condition="'$(IsClientLibrary)' == 'true'">
```

Comment:
Is there a reason you added the condition on the ruleset file?

Reply:
I guess it doesn't make a difference since the actual package reference is conditioned with ISClientLibrary, but I was trying to start with client libraries for now

Reply:
Yes since you set EnableStyleCopAnalyzers only for IsClientLibrary above I didn't know what extra this condition added.